### PR TITLE
feat(CSS): support master-node and purchase a yearly/monthly cluster

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

1. support add master, cold, client node to cluster
2. support purchase a yearly/monthly cluster.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

### Compatibility Test Cases
```
make testacc TEST='./huaweicloud/services/acceptance/css' TESTARGS='-run=TestAccCssCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run=TestAccCssCluster -timeout 360m -parallel 4
=== RUN   TestAccCssCluster_basic
=== PAUSE TestAccCssCluster_basic
=== RUN   TestAccCssCluster_security
=== PAUSE TestAccCssCluster_security
=== CONT  TestAccCssCluster_basic
=== CONT  TestAccCssCluster_security
--- PASS: TestAccCssCluster_security (952.91s)
--- PASS: TestAccCssCluster_basic (1790.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       1790.524s
```

### Test Cases
```
make testacc TEST='./huaweicloud/services/acceptance/css' TESTARGS='-run=TestAccCssCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run=TestAccCssCluster -timeout 360m -parallel 4
=== RUN   TestAccCssCluster_basic
=== PAUSE TestAccCssCluster_basic
=== RUN   TestAccCssCluster_security
=== PAUSE TestAccCssCluster_security
=== CONT  TestAccCssCluster_basic
=== CONT  TestAccCssCluster_security
--- PASS: TestAccCssCluster_security (1294.76s)
--- PASS: TestAccCssCluster_basic (1814.51s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       1814.554s
```

### CODE REVIEW

File: docs/resources/css_cluster.md [hunk 1]
```diff
@@ -83,33 +136,52 @@ The following arguments are supported:
...
+  Changing this parameter will create a new resource.
 
-The `network_info` block supports:
+* `auto_renew` - (Optional, String, ForceNew) Specifies whether auto renew is enabled.
+  Valid values are `true` and `false`, defaults to `false`.
+  Changing this parameter will create a new resource.
 
-* `vpc_id` - (Required, String, ForceNew) VPC ID, which is used for configuring cluster network. Changing this parameter
-  will create a new resource.
+* `auto_pay` - (Optional, String, ForceNew) Specifies whether auto pay is enabled.
```
Comment (niuzhenguo) [inline docs/resources/css_cluster.md]: pls get rid of `auto_pay`, we will not support this any more

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 2]
```diff
@@ -181,8 +271,100 @@ func ResourceCssCluster() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"period_unit": common.SchemaPeriodUnit(nil),
+			"period":      common.SchemaPeriod(nil),
+			"auto_renew":  common.SchemaAutoRenew(nil),
+			"auto_pay":    common.SchemaAutoPay(nil),
```
Comment (niuzhenguo) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: get rid of `auto_pay`

File: docs/resources/css_cluster.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/css_cluster.md]: ok,done

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: ok,done

File: huaweicloud/services/acceptance/css/resource_huaweicloud_css_cluster_test.go [hunk 3]
```diff
@@ -100,117 +136,114 @@ func testAccCssCluster_basic(rName string, nodeNum int, keepDays int, tag string
...
-		_, err := cluster.Get(client, rs.Primary.ID)
-		if err == nil {
-			return fmtp.Errorf("css cluster still exists, cluster_id:%s", rs.Primary.ID)
-		}
-	}
+  }
 
-	return nil
-}
+  kibana_public_access {
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/css/resource_huaweicloud_css_cluster_test.go]: don't enalbe kibana public access in test, we should avoid open service port with eip address.

File: docs/resources/css_cluster.md [hunk 4]
```diff
@@ -83,33 +136,47 @@ The following arguments are supported:
...
+The `ess_node_config`, `master_node_config`, `client_node_config` and `cold_node_config` block supports:
 
-* `subnet_id` - (Required, String, ForceNew) Subnet ID. All instances in a cluster must have the same subnet which
-  should be configured with a *DNS address*. Changing this parameter will create a new resource.
+* `flavor` - (Required, String, ForceNew) Specifies the flavor name. For example: value range of flavor ess.spec-2u8g:
+  40 GB to 800 GB, value range of flavor ess.spec-4u16g: 40 GB to 1600 GB, value range of flavor ess.spec-8u32g: 80 GB
+  to 3200 GB, value range of flavor ess.spec-16u64g: 100 GB to 6400 GB, value range of flavor ess.spec-32u128g: 100 GB
+  to 10240 GB. Changing this parameter will create a new resource.
+
+* `instance_number` - (Required, String) Specifies the number of cluster instances.
```
Comment (Jason-Zhang9309) [inline docs/resources/css_cluster.md]: the type should be int

Comment (hmbig2) [reply docs/resources/css_cluster.md]: fixed

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 5]
```diff
@@ -2,16 +2,23 @@ package css
...
 	"context"
+	"encoding/json"
 	"fmt"
+	"log"
+	"net/http"
+	"strings"
 	"time"
 
-	"github.com/chnsz/golangsdk"
 	"github.com/chnsz/golangsdk/openstack/css/v1/cluster"
```
Comment (ShiChangkuo) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: please do not import packages in golangsdk and huaweicloud-sdk-go-v3, and it's easy to remove `github.com/chnsz/golangsdk/openstack/css/v1/cluster` package.

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 6]
```diff
@@ -41,35 +48,40 @@ func ResourceCssCluster() *schema.Resource {
...
 				ForceNew: true,
 			},
 
 			"expect_node_num": {
-				Type:     schema.TypeInt,
-				Optional: true,
-				Default:  1,
+				Type:       schema.TypeInt,
+				Optional:   true,
+				Deprecated: "please use ess_node_config.instance_number instead",
```
Comment (ShiChangkuo) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: s/ess_node_config.instance_number/ess_node_config.**0**.instance_number/g

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go [hunk 7]
```diff
@@ -41,35 +48,40 @@ func ResourceCssCluster() *schema.Resource {
...
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
 
 			"expect_node_num": {
-				Type:     schema.TypeInt,
-				Optional: true,
-				Default:  1,
+				Type:       schema.TypeInt,
```
Comment (ShiChangkuo) [inline huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: suggest to align the deprecated parameters at last of schemas

File: docs/resources/css_cluster.md [hunk 8]
```diff
@@ -12,69 +12,122 @@ Manages CSS cluster resource within HuaweiCloud
...
 variable "vpc_id" {}
+variable "subnet_id" {}
+variable "secgroup_id" {}
 
-resource "huaweicloud_networking_secgroup" "secgroup" {
-  name        = "terraform_test_security_group"
-  description = "terraform security group acceptance test"
+resource "huaweicloud_css_cluster" "cluster" {
+  name            = "terraform_test_cluster"
+  engine_version  = "7.10.2"
```
Comment (ShiChangkuo) [inline docs/resources/css_cluster.md]: two spaces left on `=`

File: docs/resources/css_cluster.md [hunk 9]
```diff
@@ -12,69 +12,122 @@ Manages CSS cluster resource within HuaweiCloud
...
-* `engine_type` - (Optional, String, ForceNew) Engine type. The default value is "elasticsearch". Currently, the value
-  can only be "elasticsearch". Changing this parameter will create a new resource.
+* `name` - (Required, String, ForceNew) Specifies the cluster name. It contains 4 to 32 characters.
+  Only letters, digits, hyphens (-), and underscores (_) are allowed. The value must start with a letter.
+  Changing this parameter will create a new resource.
 
-* `engine_version` - (Required, String, ForceNew) Engine version. Versions 5.5.1, 6.2.3, 6.5.4, 7.1.1, 7.6.2 and 7.9.3
-  are supported. Changing this parameter will create a new resource.
+* `engine_type` - (Optional, String, ForceNew) Specifies the engine type. The default value is `elasticsearch`.
+  The valid values are `elasticsearch`, `logstash`. Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/css_cluster.md]: logstash is not shown in console, please check it

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: OK

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: keep

File: huaweicloud/services/css/resource_huaweicloud_css_cluster.go (hunk 7 above)
Comment (hmbig2) [reply huaweicloud/services/css/resource_huaweicloud_css_cluster.go]: OK

File: docs/resources/css_cluster.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/css_cluster.md]: remove it

File: docs/resources/css_cluster.md (hunk 8 above)
Comment (hmbig2) [reply docs/resources/css_cluster.md]: fixed

**Pull request merged**